test/conformance/ptp/ptp.go

```diff
@@ -943,3 +1045,294 @@ func ptpEventEnabled() bool {
+						},
+						{
+							Effect:            "NoExecute",
+							Key:               "node.kubernetes.io/unreachable",
+							Operator:          "Exists",
+							TolerationSeconds: &tolerationsSeconds,
+						},
+						{
+							Effect: "NoSchedule",
+							Key:    "node-role.kubernetes.io/master",
+						},
+					},
+					Volumes: []v1core.Volume{
+						{
+							Name: "host",
+							VolumeSource: v1core.VolumeSource{
+								HostPath: &v1core.HostPathVolumeSource{
+									Path: "/",
+									Type: &hostPathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	fmt.Println("Creating daemon set...")
+	_, err := client.Client.DaemonSets("default").Create(context.TODO(), rebootDaemonSet, metav1.CreateOptions{})
+	if err != nil {
+		panic(err)
+	}
+	Eventually(func() int32 {
+		daemonset, err := client.Client.DaemonSets("default").Get(context.Background(), "ptp-reboot", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		return daemonset.Status.NumberReady
+	}, 2*time.Minute, 2*time.Second).Should(Equal(nodesCount))
+
+	var ptpPods *v1core.PodList
+	Eventually(func() int {
+		ptpPods, err = client.Client.Pods("default").List(context.Background(), metav1.ListOptions{LabelSelector: "name=ptp-reboot"})
+		Expect(err).ToNot(HaveOccurred())
+		return len(ptpPods.Items)
+	}, 2*time.Minute, 2*time.Second).Should(Equal(int(nodesCount)))
+
+	return ptpPods, err
+}
+
+func rebootNode(masterNodeLabel, slaveNodeLabel string, nodesCount int32) {
+
+	// create a daemon set to get the list of ptp reboot pods
+	ptpRebootDsPods, err := createRebootDaemonSet(nodesCount)
+	pod := ptpRebootDsPods.Items[0]
+
+	countPtpRebootPods := len(ptpRebootDsPods.Items)
+
+	// Reboot the machine
+	var ptpPodsAfterReboot *v1core.PodList
+	if len(ptpRebootDsPods.Items) >= 1 {
+		Eventually(func() int {
+
+			_, err = pods.ExecCommand(client.Client, pod, "container-00", []string{"chroot", "/host", "systemctl", "reboot"})
+			Expect(err).ToNot(HaveOccurred())
+
+			time.Sleep(5 * time.Minute) // Is there any alternative, here?
+			// get ptp running pods
+			ptpPodsAfterReboot, err := client.Client.Pods(PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(ptpPodsAfterReboot.Items)).To(BeNumerically(">", 0), "linuxptp-daemon is not deployed on cluster")
+			return len(ptpPodsAfterReboot.Items)
+		}, 5*time.Minute, 10*time.Second).WithTimeout(10 * time.Minute).Should(BeNumerically(">=", countPtpRebootPods)) // expected to see the same number of ptp
+	}
+
+	// Find the ptp in slave machine
+	ptpSlaveRunningPods := []v1core.Pod{}
+	ptpMasterRunningPods := []v1core.Pod{}
+
+	var ptpRunningPods []v1core.Pod
+
+	ptpPodsAfterReboot, err = client.Client.Pods(PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+
+	for _, pod := range ptpPodsAfterReboot.Items {
```

Comment:
Should include the actual test after reboot in this funtion? or just call the reboot test before the other tests? I assume we woudl like to run reboot for the other clock types as well